esmvalcore/local.py

```diff
@@ -547,11 +494,131 @@ def get_multiproduct_filename(attributes, preproc_dir):
     filename_segments.append(
         f"{attributes['timerange'].replace('/', '-')}.nc")
 
-    outfile = os.path.join(
+    outfile = Path(
         preproc_dir,
         attributes['diagnostic'],
         attributes['variable_group'],
         '_'.join(filename_segments),
     )
 
     return outfile
+
+
+def _path2facets(path: Path, drs: str) -> dict[str, str]:
+    """Extract facets from a path using a DRS like '{facet1}/{facet2}'."""
+    keys = []
+    for key in re.findall(r"{(.*?)}", drs):
+        key = key.split('.')[0]  # Remove trailing .lower and .upper
+        keys.append(key)
+    start, end = -len(keys) - 1, -1
+    values = path.parts[start:end]
+    facets = {key: values[idx] for idx, key in enumerate(keys)}
+    return facets
+
+
+def _select_latest_version(files: list['LocalFile']) -> list['LocalFile']:
+    """Select only the latest version of files."""
+
+    def filename(file):
+        return file.name
+
+    def version(file):
+        return file.facets.get('version', '')
+
+    result = []
+    for _, group in itertools.groupby(sorted(files, key=filename),
+                                      key=filename):
+        duplicates = sorted(group, key=version)
+        latest = duplicates[-1]
+        result.append(latest)
+    return result
+
+
+def find_files(
+    *,
```

Comment:
Out of curiosity: why do we need `*` here as first argument? Are there cases where we need to allow positional arguments but not use them?

Reply:
I thought it would look cleaner if users specify the `debug` argument explicitly when calling the function.

Reply:
I thought it would look cleaner if users specify the `debug` argument explicitly when calling the function.